modules/core/src/smithy4s/schema/Schema.scala

```diff
@@ -69,6 +71,7 @@ sealed trait Schema[A]{
     case BijectionSchema(schema, bijection) => BijectionSchema(schema.transformHintsTransitively(f), bijection)
     case RefinementSchema(schema, refinement) => RefinementSchema(schema.transformHintsTransitively(f), refinement)
     case LazySchema(suspend) => LazySchema(suspend.map(_.transformHintsTransitively(f)))
+    case s: SparseSchema[a] => SparseSchema(s.underlying.transformHintsTransitively(f)).asInstanceOf[Schema[A]]
```

Comment:
>  is the lowercased a in the bracket used for something? is it a type parameter? 

It's used to bind the type parameter to a type-variable so that you can re-use it elsewhere. Often when doing pattern matching against GADTs, several members of a case class will depend on single type, and you want to re-use this type to ascribe stuff to the right-hand-side of the arrow.  

If you do 

```
case s : SparseSchema[A] 
```

then the compiler will complain because `A` is already used and it cannot guarantee that the `A` in your case is the same as the `A` of the expression you're matching. 

If you use existentials (which, in all frankness, I could do here), then it'll be hard to reconcile expressions on the RHS.

Reply:
IIRC this feature is severely half-baked in Scala 2, it can even succeed to match bogus values and result in `ClassCastException` 😓 

but if it compiles in 3.x, it should be fine on both versions